src/cr/cube/cube_slice.py

```diff
@@ -315,10 +339,31 @@ def zscore(self, weighted=True, prune=False, hs_dims=None):
         total = self.margin(weighted=weighted)
         colsum = self.margin(axis=0, weighted=weighted)
         rowsum = self.margin(axis=1, weighted=weighted)
-        return self._calculate_std_res(
+        res = self._calculate_std_res(
             counts, total, colsum, rowsum,
         )
 
+        if hs_dims:
+            res = self._intersperse_hs_in_std_res(hs_dims, res)
+
+        if prune:
+            arr = self.as_array(
+                prune=prune,
+                include_transforms_for_dims=hs_dims,
+            )
+            if isinstance(arr, np.ma.core.MaskedArray):
+                res = np.ma.masked_array(res, arr.mask)
```

Comment:
I think this whole segment that occurs in multiple places is inlining a method like `._pruning_mask()` which can be memoized for performance. After extracting that (and using 'zscores' instead of 'res'), perhaps something like: `._apply_pruning_mask()` is possible such that these 7 lines becomes simply:

```python
if prune:
    zscores = self._apply_pruning_mask(zscores, hsdims)
```

Reply:
I agree. H&S pose many problems when passed around as parameters, but I hope they'll go away eventually.

Anyhow, I don't think this PR is the best place to address this particular issue, since it originally only _moves_ the functionality to a more appropriate place to live in. I'd ticket this and move along with the work done here.

Reply:
Done